subprojects/snapshots/src/main/java/org/gradle/internal/fingerprint/impl/RelativePathFingerprintingStrategy.java

```diff
@@ -35,10 +36,12 @@
  * File names for root directories are ignored. For root files, the file name is used as normalized path.
  */
 public class RelativePathFingerprintingStrategy extends AbstractFingerprintingStrategy {
+    public static final String IDENTIFIER = "RELATIVE_PATH";
+
     private final StringInterner stringInterner;
 
     public RelativePathFingerprintingStrategy(StringInterner stringInterner) {
-        super("RELATIVE_PATH", NormalizedPathFingerprintCompareStrategy.INSTANCE);
+        super(IDENTIFIER);
```

Comment:
Why not have a singleton instance of this thing as well?

Reply:
Since the strategy needs a `StringInterner`, which is not available for a singleton.